unit_test/model_mgmt_test.cc

```diff
@@ -135,6 +139,33 @@ BOOST_AUTO_TEST_CASE(mock_azure_storage_model_data)
   scode = data_transport->get_data(md, &status);
   BOOST_CHECK_EQUAL(md.refresh_count(), 2);
 }
+
+BOOST_AUTO_TEST_CASE(mock_azure_storage_model_api_data)
+{
+    //create a simple ds configuration
+    u::configuration cc;
+    auto scode = cfg::create_from_json(JSON_CFG, cc);
+    BOOST_CHECK_EQUAL(scode, r::error_code::success);
+    http_headers header;
+    header.add(_XPLATSTR("Ocp-Apim-Subscription-Key"), "apiKey123");
+    cc.set(r::name::EH_TEST, "true"); // local test event hub
+
+    auto http_client = new mock_http_client("http://test.com");
+    std::unique_ptr<m::i_data_transport> data_transport(new m::restapi_data_transport(http_client, header, true, nullptr));
+
+    r::api_status status;
+
+    m::model_data md;
+    BOOST_CHECK_EQUAL(md.refresh_count(), 0);
+    scode = data_transport->get_data(md, &status);
+    BOOST_CHECK_EQUAL(scode, r::error_code::success);
+    BOOST_CHECK_EQUAL(md.refresh_count(), 1);
+    BOOST_CHECK_EQUAL(scode, r::error_code::success);
+    //std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+    Sleep(1000);
```

Comment:
What is the reason to sleep for 1 second? Can you add comment for this?

Reply:
refresh count will only increase if the file last modified and the current date time are different. Sometimes because of the fast processing they are same and refresh count is not incrementing. To prevent that added sleep(1000)